modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -2240,6 +2240,14 @@ private void initializeDataStorageConfiguration(IgniteConfiguration cfg) throws
 
             if (!cfg.isClientMode() && cfg.getDataStorageConfiguration() == null)
                 cfg.setDataStorageConfiguration(new DataStorageConfiguration());
+
+            if (cfg.getDataStorageConfiguration() != null &&
+                cfg.getDataStorageConfiguration().getWalPath() != null &&
+                cfg.getDataStorageConfiguration().getWalPath()
+                .equals(cfg.getDataStorageConfiguration().getWalArchivePath())) {
+                throw new IllegalArgumentException("Bad configuration. You have same paths for WAL and WAL archive. " +
```

Comment:
DataStorage configuration parameters invalid: WAL path and WAL archive path must be different.

Reply:
Wrap it with if statement.